pkg/binding/selectors.go

```diff
@@ -137,21 +134,20 @@ func (c *Controller) updateResolutions(ctx context.Context, objIdentifier util.O
 	// NOTE that this takes care of the case where the object was previously selected by a singleton binding
 	// and is no longer selected by any binding.
 	if !objBeingDeleted && !isSelectedBySingletonBinding {
-		if err := c.handleSingletonLabel(ctx, obj, objGVR, emptyBindingPolicyName); err != nil {
+		if err := c.handleSingletonLabel(ctx, obj, objGVR, util.BindingPolicyLabelSingletonStatusValueUnset); err != nil {
 			return fmt.Errorf("failed to remove singleton label from object: %w", err)
```

Comment:
This error message should be updated since the label no longer gets removed.

Reply:
Perhaps both this and the error message after the other call can be changed to directly reflect controller::handleSingletonLabel failing (this was not introduced in this PR)